src/SolverAPI.jl

```diff
@@ -558,3 +563,79 @@ function solveCliqUp!(fg::AbstractDFG,
 
   return hist, beliefMessageOut
 end
+
+"""
+Internal function used for solveCliqDown!
+"""
+function _buildMessageDown(fg, tree, cliqid)
+  cliq = getClique(tree, cliqid)
+  parent = getParent(tree, cliq)[1]
+  return getCliqDownMsgsAfterDownSolve(fg, parent; status=DOWNSOLVED)
```

Comment:
what if the clique has status `MARGINALIZED`, then `DOWNSOLVED` is not entirely correct.  Is there perhaps an easy way to use the actual clique status vs hardcoded?

Reply:
or via keyword to allow user to override the default status?

Reply:
also e.g.:
https://github.com/JuliaRobotics/IncrementalInference.jl/pull/1186/files#diff-bdbb86421fa7a813967257eb8845a821e8595683ead3f4615750646db435d942R640

Reply:
There is no parent clique, so no easy way. 
We can do kwarg, but does it matter? Downsolve is treated the same in all cases.

Reply:
i'd suggest just going with kwarg to make the hard coded a little better.  We'll see if tree CSM logic evolves to a point where DOWNSOLVE vs MARGINALIZED or others make a difference -- agreed not 100% yet today, but I have reason to suspect things will change there over time.